stop/stop.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016 CoreOS, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stop implements a pattern for shutting down a group of processes.
+package stop
+
+import (
+	"sync"
+)
+
+// AlreadyDone is a constant used to express that a process has already been
```

Comment:
what's the point of this? it's not used anywhere, and I can't see what it'd be used for anyway